### PR TITLE
Add condition granting support to AutoTarget.

### PR DESCRIPTION
**Pull request by pchote**
This allows traits to be toggled based on the current actor stance.  This can be used by mods to draw a stance decoration, and gives us the first half of the logic we need to properly implement the original game's autotarget behaviour.

As a testcase, this positions the health decoration on TD infantry in one of four corners based on the current stance.  This commit should obviously be removed before merging.

Supersedes #13275.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/AutoTarget.cs [hunk 1]
```diff
@@ -35,6 +35,22 @@ public class AutoTargetInfo : ConditionalTraitInfo, Requires<AttackBaseInfo>, Us
...
+		[GrantedConditionReference]
+		[Desc("The condition to grant to self while in the HoldFire stance.")]
+		public readonly string HoldFireCondition = null;
+
+		[GrantedConditionReference]
+		[Desc("The condition to grant to self while in the ReturnFire stance.")]
+		public readonly string ReturnFireCondition = null;
+
+		[GrantedConditionReference]
+		[Desc("The condition to grant to self while in the HoldFire stance.")]
```
Comment (rob-v) [inline OpenRA.Mods.Common/Traits/AutoTarget.cs]: ...Defend stance.

File: OpenRA.Mods.Common/Traits/AutoTarget.cs [hunk 2]
```diff
@@ -35,6 +35,22 @@ public class AutoTargetInfo : ConditionalTraitInfo, Requires<AttackBaseInfo>, Us
...
+		public readonly string DefendCondition = null;
+
+		[GrantedConditionReference]
+		[Desc("The condition to grant to self while in the HoldFire stance.")]
```
Comment (rob-v) [inline OpenRA.Mods.Common/Traits/AutoTarget.cs]: ...AttackAnything stance.

**Comment by pchote** [comment]
Fixed.  I have also removed the testcase commit, if you want to try it yourself add the following to the cnc `^Soldier` default:
```
AutoTarget:
	HoldFireCondition: stance-holdfire
	ReturnFireCondition: stance-returnfire
	DefendCondition: stance-defend
	AttackAnythingCondition: stance-attackanything
WithDecoration@HOLDFIRE:
	Image: pips
	Sequence: pip-heal
	ReferencePoint: Top, Left
	RequiresCondition: stance-holdfire
WithDecoration@RETURNFIRE:
	Image: pips
	Sequence: pip-heal
	ReferencePoint: Top, Right
	RequiresCondition: stance-returnfire
WithDecoration@DEFEND:
	Image: pips
	Sequence: pip-heal
	ReferencePoint: Bottom, Right
	RequiresCondition: stance-defend
WithDecoration@ATTACKANYTHING:
	Image: pips
	Sequence: pip-heal
	ReferencePoint: Bottom, Left
	RequiresCondition: stance-attackanything
```

**Comment by rob-v** [comment]
:1 Thanks, interestering test.

**Review by rob-v — APPROVED** [review]

**Review by GraionDilach — APPROVED** [review]
:+1:

**Comment by pchote** [comment]
Superseded by #13389.

**Pull request closed without merging**